### DISCUSSION TITLE
General File System Help

### DISCUSSION THREAD

**Discussion by deepanshug** [winfsp/winfsp#456] [General]
I have been trying to build a file system using WinFSP where
1.	I am opening a MS office (e.g. word.docx) file but changing the file descriptor to point to some other file. (word.txt file) in Open function
2.	When read happens, it happens correctly from word.txt file and displays the content in my word.docx file.
3.	When I write to it, expectation is that it should write to word.txt file and should not change any content in word.docx.

Note: 
1.	word.docx file is just a container, actual read/write should happen in word.txt file. Here read works fine but write is not working for office files. The issue here is because of temporary files being generated by ms office but how to handle this is still not known.
2.	Auto save option is turned off for time being.

Tried so far:
When saving data, word creates a “tmp” file, I changed the descriptor again to “txt” file in Open function but write function again receives a descriptor to tmp file. 
Changed the file descriptor in “Write” function to point to txt file but again it writes to docx file. IN addition it deletes my txt file as well.

I am totally clueless as what’s happening here when writing to docx file. 
Is there any pointers you can share to my problem please.

Thanks 
Deepanshu

**Comment by Masamune3210** [comment]
Think you are running into Word behavior more than anything else, when you open a file it creates a tmp file, which IIRC is not just a simple text file, and then when it finalizes saving the file, the last few steps are dealing with that tmp file, which ends up in the end deleting the tmp file since the contents have at that point been validated to have been correctly saved into the docx
